### ISSUE TITLE
Add capability to create sql resources from queries

### ISSUE THREAD

**Issue by erik-meko** [dlt-hub/dlt#1997]
### Feature description

Create a sql resource in `dlt.sources.sql_database` that allows creating models/tables with sql. Could be named sql_query and work like sql_table, except that instead of creating a resource given a single table, it can use multiple database objects as input, preferably defined using a sql statement.

This is useful when the source database is not in the consumers control, so that it is not possible to create a static view

### Are you a dlt user?

I'd consider using dlt, but it's lacking a feature I need.

### Use case

I have a need to load incremental tables where the cursor is in a second table, i.e. I would like to do

```
SELECT s.*, m.updatedtimestamp
FROM supporting_table AS s
JOIN main_table AS m ON s.main_table_id = m.id
WHERE m.updatedtimestamp >= :_last_updated
```

### Proposed solution

Add a new "resource type" in sql_database, that allows to use more ephemeral entities like select statements (or temp tables)

**Comment by erik-meko** [comment]
I am not allowed to modify the source and unfortunately I wasnt successful in my attempts to apply a query_adapter_callback (I failed to understand how to modify the schema) Alternative 3 looks simple enough, however, I dropped the PoC after trying to implement another pipeline in self-hosted airflow with azure keyvault as secrets backend. 

During the run, an api pipeline is apparently looking for a lot of airflow variables. With a failing remote call to the azure keyvault, I got ~25 lines of stacktrace in the log for each, totalling in many 100 lines of stacktrace in the airflow log. Troubleshooting it would have been a nightmare and since this part wasnt well documented, I shut down the PoC.

**Comment by rudolfix** [comment]
sorry to hear that `dlt` didn't work for you. airflow config provider can be disabled in config toml or via env. variable. there's not much we can do for Airflow to not log this, we already use Variable.get method with a default value to get secrets. it seems Airflow decides to log that anyway.

**Comment by rudolfix** [comment]
actually free text sql support is coming here #2070

**Comment by rudolfix** [comment]
fixed in 1.4.1
